tests/core/resource/test_viewset.py

```diff
@@ -355,18 +349,26 @@ def test_is_endpoint_enabled_returns_true_for_head_if_readonly(self):
         self.assertTrue(is_enabled)
 
 
-class TestPartialSchemaTest(unittest.TestCase):
+class TestViewsetSchemasTest(unittest.TestCase):
 
     def test_partial_schema_ignores_unknown(self):
         schema = PartialSchema()
         result = schema.deserialize({'foo': 'bar'})
         self.assertEquals(result, {})
 
+    def test_strict_schema_raise_on_unknown(self):
+        schema = StrictSchema()
+        self.assertRaises(colander.Invalid, schema.deserialize, {'foo': 'bar'})
+
 
 class ShareableViewSetTest(unittest.TestCase):
+
     def test_permission_dynamic_is_set_by_default(self):
         viewset = ShareableViewSet()
-        args = viewset.collection_arguments(mock.sentinel.resource, 'GET')
+        resource = mock.sentinel.resource
+        resource.schema = ResourceSchema
+        resource.permissions = ('read', 'write')
+        args = viewset.collection_arguments(resource, 'GET')
```

Comment:
It failed on the second bind as one call from colander that inserts a node would not accept a mock object, so I've set the `ResourceSchema` manually. I'll try to find a better approach tough.